lib/sqlalchemy/sql/schema.py

```diff
@@ -3672,6 +3689,15 @@ def __init__(
         self._column_flag = _column_flag
         self._columns = DedupeColumnCollection()
 
+        # Collect any periods objects (can't do anything about strings)
+        self._periods: Properties[Period] = Properties()
```

Comment:
mmmmm...... I still don't want Period being collected anywhere except that one place on a Table.    I would assume this is for PrimaryKeyConstraint.  PKC already refers to the Table and Period refers to Column objects in that table which have a "primary_key" attribute that is set.  So if i want the Period object that's associated with a primary key, I do:

```py
[p for p in self.table.periods if p.start.primary_key or p.end.primary_key]
```

the way it is now, the structure has a redundancy in it which also has many ways that it could be in an inconsistent state.  that's all testing / complexity overhead that isn't necessary.